Feat/profile/create

Harrish92 opened the pull request:
### Add Create Profile Feature

**Description:**
This pull request introduces the `CreateProfile` feature, allowing users to create a profile with a profile picture, name, email, age, and description. The profile picture can be selected from the gallery and displayed in a round frame.

**Changes:**
- Added `CreateProfile` composable to handle profile creation.
- Integrated image picker to select profile pictures from the gallery.
- Displayed selected profile picture in a round frame using `GlideImage`.
- Updated UI elements to include form fields for name, email, age, and description.

**Files Added/Modified:**
- `app/src/main/java/com/android/periodpals/ui/profile/CreateProfile.kt`: Added the `CreateProfile` composable.
- `app/src/main/java/com/android/periodpals/model/user/User.kt`: Existing user model for the profile creation.

**Dependencies Added:**
- `com.github.skydoves:landscapist-glide:1.5.2`
- `com.github.dhaval2404:imagepicker:2.1`

**Testing:**
- Verified that the profile image, form fields, and save button are displayed correctly.
- Ensured that the image picker launches and the selected image is displayed in a round frame.

**Screenshots:**

![photo_2024-10-10_14-02-58](https://github.com/user-attachments/assets/6955b500-b656-45a3-8f81-7a542350289d)

**Notes:**
- Ensure to update the dependencies in the `build.gradle` file before testing.

francelu reviewed (changes requested):
### First of all
I love the hippopotamus profile picture!

### Important
Please ensure to run the following on all files you’ve modified:
- **Optimize Imports**
- **Reformat Code**
- **Code Cleanup**
- **Run `gradle ktfmt`**

### File Review
- **`app/build.gradle.kts`**: LGTM.
  
- **`AndroidManifest.xml`**: LGTM.
  
- **`MainActivity.kt`**: 
  - Keep `CountriesList()` and remove unnecessary imports. Please run **Optimize Imports**.
  - Ensure to run **ktfmt** on this file as well.

- **`User.kt`**: 
  - Remove the `{}` at the end of the `data class` definition.

- **`CreateProfile.kt`**:
  - Ensure to **Optimize Imports**, **Reformat Code**, perform **Code Cleanup**, and run **ktfmt**.
  - The input field labelled "Age" may cause confusion, as it is unclear whether the user is expected to enter their age or their date of birth. It would be beneficial to rename this field to "Date of Birth" to provide clarity. Additionally, please specify the required format (e.g., "Enter your date of birth in DD/MM/YYYY format") to guide users effectively.

- **`Theme.kt`**: 
  - Consider uncommenting the other colours for overriding. These will likely be redefined in Sprint 2.

- **`generic_avatar.png`**: LGTM.  
- **`roboto.xml`**: 
  - I don’t see the purpose of this file, as it doesn’t seem to be referenced anywhere. Could you clarify its necessity for the project?

- **`libs.versions.toml`**: LGTM. 

### Additional Notes
- **Missing Tests**: It appears that UI tests for the Create Profile screen are missing. Please refer to the tests for `AddToDoScreen` in the bootcamp for guidance on how to implement these.

francelu reviewed (commented):
### Improvements
Please address the following points real quick:

1. **Date Validation Logic**:
   - The date validation logic should be extracted into a separate function to enhance readability and facilitate easier testing.

2. **Test Assertions**:
   - Add assertions in `testSaveButtonClickWithValidDate` and `testSaveButtonClickWithInvalidDate` to verify the application's behavior after clicking the save button. This will ensure that the date input is handled correctly and that appropriate feedback is provided to the user.  

Otherwise, LGTM.

Harrish92 commented:
### Improvements

I have addressed the following points:

#### Date Validation Logic:
The date validation logic has been extracted into a separate function to enhance readability and facilitate easier testing.

#### Test Assertions:
Added assertions in `testSaveButtonClickWithValidDate` and `testSaveButtonClickWithInvalidDate` to verify the application's behavior after clicking the save button. This ensures that the date input is handled correctly and that appropriate feedback is provided to the user.

francelu commented on app/src/main/AndroidManifest.xml:
```diff
@@ -15,7 +15,7 @@
         tools:targetApi="31">
         <activity
-            android:name=".MainActivity"
+            android:name="com.android.periodpals.MainActivity"
```
You don't need to specify `com.android.periodpals.MainActivity`. In the authentication UI branch, I ran `MainActivity` with just `.MainActivity` while the emulator was on.  
Your version may work as well, so let’s stick with it for now. :))

francelu commented on app/src/main/java/com/android/periodpals/MainActivity.kt:
```diff
@@ -30,7 +32,8 @@ class MainActivity : ComponentActivity() {
         // A surface container using the 'background' color from the theme
         Surface(modifier = Modifier.fillMaxSize(), color = MaterialTheme.colorScheme.background) {
-          CountriesList()
+          // CountriesList()
```
It’s better to keep CountriesList() here to avoid merging incomplete screens into the main branch.  
Once we merge the navigation, this issue will be resolved. For now, let's leave it as is.

francelu reviewed (approved):
LGTM. :)

coaguila reviewed (approved):
LGTM
